src/extension.ts

```diff
@@ -748,20 +766,37 @@ export class Extension implements RunHooks {
     return this._reusedBrowser.browserServerWSEndpoint();
   }
 
-  private _showTrace(testItem: vscodeTypes.TestItem) {
+  private async _showTrace(testItem: vscodeTypes.TestItem) {
     const traceUrl = (testItem as any)[traceUrlSymbol];
     const testModel = this._models.selectedModel();
     if (testModel)
-      this._traceViewer.open(traceUrl, testModel.config);
+      this._traceViewer.open(traceUrl, await this._extendTestConfig(testModel));
   }
 
-  private _treeItemSelected(treeItem: vscodeTypes.TreeItem | null) {
+  private async _treeItemSelected(treeItem: vscodeTypes.TreeItem | null) {
```

Comment:
Making event handlers async is not nice since we end up with unhandled rejections. Let's try to keep things sync.